File: php/features.php
<?php
declare(strict_types=1);

// VisitorFrequenciesLucerneCh SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class VisitorFrequenciesLucerneChFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new VisitorFrequenciesLucerneChBaseFeature();
            case "test":
                return new VisitorFrequenciesLucerneChTestFeature();
            default:
                return new VisitorFrequenciesLucerneChBaseFeature();
        }
    }
}

File: py/features.py
# VisitorFrequenciesLucerneCh SDK feature factory

from feature.base_feature import VisitorFrequenciesLucerneChBaseFeature
from feature.test_feature import VisitorFrequenciesLucerneChTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VisitorFrequenciesLucerneChBaseFeature(),
        "test": lambda: VisitorFrequenciesLucerneChTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
